SwordHandler.inc.php

```diff
@@ -0,0 +1,122 @@
+	/** @var SwordPlugin Sword plugin */
+	protected $_parentPlugin = null;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		parent::__construct();
+		// set reference to markup plugin
+		$this->_parentPlugin = PluginRegistry::getPlugin('generic', 'swordplugin');
+		$this->addRoleAssignment(
+			array(ROLE_ID_MANAGER),
+			array('depositPoints','performManagerOnlyDeposit')
+		);
+		$this->addRoleAssignment(
+			array(ROLE_ID_MANAGER, ROLE_ID_AUTHOR),
+			array('index')
+		);
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	public function authorize($request, &$args, $roleAssignments) {
+		$op = $request->getRouter()->getRequestedOp($request);
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Get reference to the sword plugin
+	 * @return SwordPlugin
+	 */
+	public function getSwordPlugin() {
+		return $this->_parentPlugin;
+	}
+
+	/**
+	 * Returns deposit point details
+	 * @param $args array
+	 * @param $request PKPRequest
+	 *
+	 * @return JSONMessage
+	 */
+	public function depositPoints($args, $request) {
+		$context = $request->getContext();
+		$depositPointId = $request->getUserVar('depositPointId');
+		$this->getSwordPlugin()->import('classes.DepositPoint');
+		$depositPointDao = DAORegistry::getDAO('DepositPointDAO');
+		$depositPoint = $depositPointDao->getById($depositPointId, $context->getId());
+		if (!$depositPoint) {
+			return new JSONMessage(false);
+		}
+
+		$this->getSwordPlugin()->import('classes.DepositPointsHelper');
+		$collections = DepositPointsHelper::loadCollectionsFromServer(
+			$depositPoint->getSwordUrl(),
+			$depositPoint->getSwordUsername(),
+			$depositPoint->getSwordPassword()
+		);
+		return new JSONMessage(true, array(
+			'username' => $depositPoint->getSwordUsername(),
+			'password' => SWORD_PASSWORD_SLUG,
+			'depositPoints' => $collections,
+		));
+	}
+
+	/**
+	 * Returns deposit point details
+	 * @param $args array
+	 * @param $request PKPRequest
+	 *
+	 * @return JSONMessage
+	 */
+	public function index($args, $request) {
+		$context = $request->getContext();
+		$user = $request->getUser();
+		$articleId = (int) array_shift($args);
+		$save = array_shift($args) == 'save';
+
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($articleId);
+
+		if (!$submission || !$user || !$context ||
+			($submission->getPrimaryAuthor()->getEmail() != $user->getEmail()) ||	// TODO is this the best way to port ($article->getUserId() != $user->getId()) to OJS3?
+			($submission->getContextId() != $context->getId())) {
+				$request->redirect(null, 'index');
+		}
+
+		$swordPlugin = $this->getSwordPlugin();
+		$swordPlugin->import('AuthorDepositForm');
+		$authorDepositForm = new AuthorDepositForm($swordPlugin, $context, $submission);
+
+		if ($save) {
+			$authorDepositForm->readInputData();
+			if ($authorDepositForm->validate()) {
+				$authorDepositForm->execute($request);
+				$request->redirect(null, 'submissions');
```

Comment:
The `$request->redirect` function ends execution, so the below `initData`/`display` calls are needlessly duplicated. You can just end off your execution blocks and have one set of `initData`/`display` calls at the end of the function.

Reply:
refactored.